qutip/solver/flimesolve.py

```diff
@@ -59,107 +59,103 @@ def _floquet_rate_matrix(floquet_basis,
     '''
     Hdim = len(floquet_basis.e_quasi)
 
-    time = T
+    time = floquet_basis.T
     dt = time / Nt
     tlist = np.linspace(0, time - dt, Nt)
+    omega = 2 * np.pi / time
 
     '''
     First,divide all quasienergies by omega to get everything in terms of
         omega.
     '''
 
-    def delta(a, ap, b, bp, l, lp):
-        return ((floquet_basis.e_quasi[a] - floquet_basis.e_quasi[ap])
-                - (floquet_basis.e_quasi[b] - floquet_basis.e_quasi[bp])) \
-            / (list(omega.values())[0]) \
-            + (l - lp)
+    delta_m = np.add.outer(floquet_basis.e_quasi, -floquet_basis.e_quasi)
+    delta_m = np.add.outer(delta_m, -delta_m)
+    delta_m /= omega
```

Comment:
Computing `delta` at once instead of one per one more efficient.
`np.add.outer` is an vectorized `out[a, b] = left[a] + right[b]`.
It's not easy to find in the documentation, but really useful.

Reply:
On second look, your version is much more elegant and lends itself to the truncated 'l-k' stuff mentioned later. I'll replace mine with this.